### ISSUE TITLE
MissingOutputException when using a remote directory as an output with S3

### ISSUE THREAD

**Issue by keller-mark** [snakemake/snakemake#870]
**Snakemake version**
5.32.0

**Describe the bug**
When a rule `output:` points to a file or directory on S3, the workflow throws a `MissingOutputException`. However, I can see that the directories and files have been uploaded to my S3 bucket.

![Screen Shot 2021-02-05 at 1 20 21 PM](https://user-images.githubusercontent.com/7525285/107073121-efce9700-67b4-11eb-8d9e-57fd930c8d4b.png)

**Logs**
<details>
<summary>Logs</summary>

<pre>
Building DAG of jobs...
Using shell: /bin/bash
Provided cores: 1 (use --cores to define parallelism)
Rules claiming more threads will be scaled down.
Job counts:
	count	jobs
	1	upload_to_s3
	1
Select jobs to execute...

[Fri Feb  5 13:19:26 2021]
rule upload_to_s3:
    input: data/processed/my_data.zarr
    output: vitessce-export-examples/my_snakemake_outputs/my_data.zarr
    jobid: 0

upload: data/processed/my_data.zarr/foo/bar/baz/0.3 to s3://vitessce-export-examples/my_snakemake_outputs/my_data.zarr/foo/bar/baz/0.3
upload: data/processed/my_data.zarr/foo/bar/baz/.zarray to s3://vitessce-export-examples/my_snakemake_outputs/my_data.zarr/foo/bar/baz/.zarray
upload: data/processed/my_data.zarr/foo/bar/.zgroup to s3://vitessce-export-examples/my_snakemake_outputs/my_data.zarr/foo/bar/.zgroup
upload: data/processed/my_data.zarr/foo/bar/baz/1.1 to s3://vitessce-export-examples/my_snakemake_outputs/my_data.zarr/foo/bar/baz/1.1
upload: data/processed/my_data.zarr/foo/bar/baz/0.1 to s3://vitessce-export-examples/my_snakemake_outputs/my_data.zarr/foo/bar/baz/0.1
upload: data/processed/my_data.zarr/foo/.zgroup to s3://vitessce-export-examples/my_snakemake_outputs/my_data.zarr/foo/.zgroup
upload: data/processed/my_data.zarr/foo/bar/baz/1.0 to s3://vitessce-export-examples/my_snakemake_outputs/my_data.zarr/foo/bar/baz/1.0
upload: data/processed/my_data.zarr/foo/bar/baz/0.0 to s3://vitessce-export-examples/my_snakemake_outputs/my_data.zarr/foo/bar/baz/0.0
upload: data/processed/my_data.zarr/.zgroup to s3://vitessce-export-examples/my_snakemake_outputs/my_data.zarr/.zgroup
upload: data/processed/my_data.zarr/foo/bar/baz/0.2 to s3://vitessce-export-examples/my_snakemake_outputs/my_data.zarr/foo/bar/baz/0.2
upload: data/processed/my_data.zarr/foo/bar/baz/1.3 to s3://vitessce-export-examples/my_snakemake_outputs/my_data.zarr/foo/bar/baz/1.3
upload: data/processed/my_data.zarr/foo/bar/baz/1.2 to s3://vitessce-export-examples/my_snakemake_outputs/my_data.zarr/foo/bar/baz/1.2
Waiting at most 5 seconds for missing files.
MissingOutputException in line 14 of /Users/mkeller/research/dbmi/vitessce/vitessce-python/demos/Snakefile:
Job Missing files after 5 seconds:
vitessce-export-examples/my_snakemake_outputs/my_data.zarr
This might be due to filesystem latency. If that is the case, consider to increase the wait time with --latency-wait.
Job id: 0 completed successfully, but some output files are missing. 0
  File "/Users/mkeller/anaconda3/envs/vitessce-python-dev/lib/python3.8/site-packages/snakemake/executors/__init__.py", line 583, in handle_job_success
  File "/Users/mkeller/anaconda3/envs/vitessce-python-dev/lib/python3.8/site-packages/snakemake/executors/__init__.py", line 252, in handle_job_success
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: /Users/mkeller/research/dbmi/vitessce/vitessce-python/demos/.snakemake/log/2021-02-05T131926.089696.snakemake.log
</pre>
</details>

**Minimal example**
```py
import zarr
import numpy as np
from pathlib import Path, PosixPath
from snakemake.remote.S3 import RemoteProvider as S3RemoteProvider

S3 = S3RemoteProvider()

DATA_DIR = Path("data")
PROCESSED_DIR = DATA_DIR / "processed"

BUCKET_NAME = "vitessce-export-examples"
BUCKET = PosixPath(BUCKET_NAME) / "my_snakemake_outputs"

rule upload_to_s3:
    input:
        (PROCESSED_DIR / "my_data.zarr")
    output:
        S3.remote(directory(str(BUCKET / "my_data.zarr")))
    shell:
        """
        aws s3 cp --recursive {input} s3://{output}
        """

rule create_local_file:
    output:
        directory(PROCESSED_DIR / "my_data.zarr")
    run:
        root = zarr.open(output[0], mode='w')
        z = root.zeros('foo/bar/baz', shape=(100,200), chunks=(50, 50), dtype='i4')
        z[:] = 99
```

**Additional context**
Possibly related to #396 and #576

**Comment by jsadler2** [comment]
I'm trying the same thing with Zarr and I have the same error

**Comment by SichongP** [comment]
Yep, same issue as https://github.com/snakemake/snakemake/issues/815
